xla/service/cpu/runtime_single_threaded_fft.cc

```diff
@@ -22,7 +22,8 @@ ABSL_ATTRIBUTE_NO_SANITIZE_MEMORY void __xla_cpu_runtime_EigenSingleThreadedFft(
     const void* run_options_ptr, void* out, void* operand, int32_t fft_type,
```

Comment:
Perhaps we should change the symbol name for this function.

Awaiting advice on deprecating v.s. maintaining the Eigen-based impl

Reply:
Just concerned about backwards compat. 

If the emitted llvm IR invokes the fn_name `runtime::kEigenSingleThreadedFftSymbolName`, as is would prevent a breaking change.

I guess what you say suggests that we do not rename to `runtime::kDuccSingleThreadedFftSymbolName`, even if slighly misleading.

Reply:
Breaking change would be better than misleading IMO.

Or you wait until Eigen allows ducc FFT, and continue to use Eigen.  I don't have a timeline for this yet though.  Or you can help  with the Eigen update.  I started a draft for using DUCC0 for matrix FFTs [here](https://gitlab.com/libeigen/eigen/-/merge_requests/1068).  The tensor module change will need a significant refactor though to better support real-valued inputs (i.e. `rfft`) and a plugin architecture like the matrix version.

Reply:
So, to be clear, this means that the Eigen interface is still the same, but you would compile Eigen with the plugin? Or would you build eigen and the plugin separately?

Reply:
The timeline seems too long as opposed to the relatively simple changes here. So my preference would be to replace here for now so we get the better CPU FFTs soon.

Reply:
> So, to be clear, this means that the Eigen interface is still the same, but you would compile Eigen with the plugin?

The Eigen interface API may change a bit, but yeah, it would still be "Eigen", so the symbol name makes sense.  And yes, you would configure Eigen to use the ducc plugin, so there would be no mention of ducc here.

> The timeline seems too long as opposed to the relatively simple changes here. So my preference would be to replace here for now.

Sure.  If you want this in long-term, then I suggest changing the symbol name, since it will not be the Eigen implementation.  You'll want to change the name anyways eventually, once it is no longer single-threaded.

Reply:
I've changed the symbol name.